File: Searching Algorithm/BinarySearch.java

import java.util.Scanner;
public class BinarySearch
{
	public static void main(String args[])
	{	
		int arr[] = {2,3,4,10,40};
		//int key = 4;
		int last = arr.length-1;
		
		// input value taking from console
		Scanner sc = new Scanner(System.in);							
		System.out.println("Enter the  Element to search: ");			
		int key = sc.nextInt();
		
		int result = binarySearch(arr, 0, last, key);
		if(result == -1)
		{
			System.out.println("Element not present in array");
		}
		else
		{
			System.out.println("Element found at index:" +result);
		}
	}
	
	public static int binarySearch(int arr[], int first, int last, int key)
	{
		if(last >= first)
		{
			int mid = first+(last - first)/2;
			if(arr[mid] == key)
			{
				return mid;
			}
			if(arr[mid] > key)
			{
				return binarySearch(arr, first, mid-1, key);
			}
			else
			{
				return binarySearch(arr, mid+1,last, key);
			}
		}
		return -1;
	}
	
}
	
	
	
	
	
	
	
	

File: Searching Algorithm/BinarySearch.c
//Hemant Sirsat

#include<stdio.h>
long int arr[100000];

long int BinarySearch(long int arr[],long int first,long int last,long int e) 
{ 
    if (last >= first) 
    { 
        long int mid = first + (last - 1) / 2; 
        if (arr[mid] == e) 
            return 1; //if element found return bool 1

        if (arr[mid] > e) 
            return BinarySearch(arr, first, mid - 1, e); //if element is smaller tha element at center of array

        return BinarySearch(arr, mid + 1, last, e); //if element is greater tha element at center of array
    }
    return 0; 
} 

void main()
{
    long int n;
    scanf("%ld",&n); //Enter Number of Elements in Array
    for(long int i = 0;i<n;i++)
        scanf("%ld",&arr[i]);  //Enter elemets in array
    long int e;                         
    printf("\nEnter Element to Search: ");
    scanf("%ld",&e);
    if(BinarySearch(arr,0,e,n))
        printf("\nElement Present");
    else printf("\nElement not found");
}
